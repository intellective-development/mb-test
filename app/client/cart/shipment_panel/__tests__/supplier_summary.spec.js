
import * as React from 'react';
import dm_factory from '@minibar/store-business/src/delivery_method/__tests__/delivery_method.factory';
import supplier_factory from '@minibar/store-business/src/supplier/__tests__/supplier.factory';

import TestProvider from 'store/views/__tests__/utils/TestProvider';
import SupplierSummary from '../supplier_summary';

describe('SupplierSummary', () => {
  it('renders', () => {
    const delivery_method = dm_factory.build();
    const supplier = supplier_factory.build({delivery_methods: [delivery_method]});
    const shipment = { supplier };

    expect(render(
      <TestProvider>
        <SupplierSummary shipment={shipment} />
      </TestProvider>
    )).toMatchSnapshot();
  });

  it('renders for pickup shipment', () => {
    const delivery_method = dm_factory.build('pickup');
    const supplier = supplier_factory.build({delivery_methods: [delivery_method]});
    const shipment = { supplier };

    expect(render(
      <TestProvider>
        <SupplierSummary shipment={shipment} />
      </TestProvider>
    )).toMatchSnapshot();
  });
});
