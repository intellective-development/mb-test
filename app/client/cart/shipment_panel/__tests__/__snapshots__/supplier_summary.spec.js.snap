// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupplierSummary renders 1`] = `
<tr
  class="shipment-table__row"
>
  <td
    colspan="5"
  >
    <div
      class="shipment-table__content-container"
    >
      <h4
        class="elMBText elMBText_ResetSpacing sp_supplier__name"
      >
        Manhattan Wine
      </h4>
      <div
        class="el-touchable el-touchable--disabled sp_supplier__container"
        role="button"
        tabindex="-1"
      >
        <h3
          class="elMBText elMBText_ResetSpacing sp_supplier__location"
        >
           123 Jump Street, New York, New York
        </h3>
      </div>
    </div>
  </td>
</tr>
`;

exports[`SupplierSummary renders for pickup shipment 1`] = `
<tr
  class="shipment-table__row"
>
  <td
    colspan="5"
  >
    <div
      class="shipment-table__content-container"
    >
      <h4
        class="elMBText elMBText_ResetSpacing sp_supplier__name"
      >
        Manhattan Wine
      </h4>
      <div
        class="el-touchable sp_supplier__container"
        role="button"
        tabindex="-1"
      >
        <h3
          class="elMBText elMBText_ResetSpacing sp_supplier__location"
        >
           123 Jump Street, New York, New York
          <span
            class="elMBText elMBText_ResetSpacing"
          >
            ãƒ»1.1 mi
          </span>
        </h3>
      </div>
    </div>
  </td>
</tr>
`;
